source/material_model/entropy_model.cc

```diff
@@ -118,22 +112,119 @@ namespace aspect
 
 
 
+    template <int dim>
+    double
+    EntropyModel<dim>::
+    equilibrate_temperature (const std::vector<double> &temperature,
+                             const std::vector<double> &mass_fractions,
+                             const std::vector<double> &entropy,
+                             const std::vector<double> &specific_heat,
+                             const double pressure,
+                             std::vector<double> &component_equilibrated_S
+                            ) const
+    {
+      AssertThrow(material_file_names.size() == temperature.size() && temperature.size() == mass_fractions.size() &&  temperature.size() == entropy.size() && temperature.size() == specific_heat.size(),
+                  ExcMessage("The temperature, chemical composition, entropy and specific heat capacity vectors"
+                             " must all have the same size as the number of look-up tables."));
+
+      std::vector<double> component_entropies  = entropy;
+      std::vector<double> component_temperatures  = temperature;
+      std::vector<double> component_heat_capacities = specific_heat;
+
+      bool equilibration = false;
+      unsigned int iteration = 0;
+      double ln_equilibrated_T = 0;
+
+      // We do the following thermodynamic iteration to find the equilibrated temperature of our components
+      // while conserving the total entropy.
+
+      while (equilibration == false)
+        {
+          Assert(iteration <= (multicomponent_max_iteration),
```

Comment:
since you now start the iteration with 0 this needs to be:
```suggestion
          Assert(iteration < multicomponent_max_iteration,
```